examples/example_pp.py

```diff
@@ -0,0 +1,274 @@
+        world_mesh = torch.distributed.device_mesh.init_device_mesh(
+            "cuda",
+            (pp_degree, dp_degree, ep_degree),
+            mesh_dim_names=(
+                "pp",
+                "dp",
+                "ep",
+            ),
+        )
+        device = torch.device("cuda")
+        pp_rank = rank
+
+    mesh = world_mesh["dp", "ep"]
+
+    global_batch_size = 4 * mesh.shape[1] * mesh.shape[2]
+    local_batch_size = global_batch_size // dp_degree
+    n_microbatches = 4
+    microbatch_size = local_batch_size // n_microbatches
+    assert (
+        microbatch_size >= 1
+    ), f"invalid config {local_batch_size=}, {n_microbatches=}"
+    spmd_batch_size = microbatch_size * dp_degree
+
+    seq_len = 1024
+    vocab_size = 102400
+    dim = 2048
+
+    def model_fn():
+        config = DeepSeekV3ModelArgs(
+            vocab_size=102400,
+            max_seq_len=seq_len,
+            dim=2048,
+            inter_dim=10944,
+            moe_inter_dim=1408,
+            n_layers=1,  # 27,
+            n_dense_layers=0,  # 1,
+            n_heads=16,
+            moe_args=MoEArgs(
+                num_experts=64,
+                num_shared_experts=2,
+                top_k=6,
+                score_func="softmax",
+                route_norm=False,
+                score_before_experts=False,
+                mesh=mesh,
+            ),
+            q_lora_rank=0,
+            kv_lora_rank=512,
+            qk_nope_head_dim=128,
+            qk_rope_head_dim=64,
+            v_head_dim=128,
+            mscale=0.70,
+            use_flex_attn=False,
+            attn_mask_type="causal",
+        )
+
+        m = DeepSeekV3Model(config).bfloat16()
+        return m
+
+    # def input_fn():
+    #     bs = spmd_batch_size
+    #     if pp_rank == 0:
+    #         x = torch.randint(
+    #             0,
+    #             vocab_size,
+    #             (bs, seq_len),
+    #             device=device,
+    #         )
+    #     else:
+    #         x = torch.randn(
+    #             (bs, seq_len, dim),
+    #             device=device,
+    #             dtype=torch.bfloat16,
+    #             requires_grad=True,
+    #         )
+    #     return x
+
+    def input_fn():
+        bs = spmd_batch_size
+        return torch.randint(
+            0,
+            vocab_size,
+            (bs, seq_len),
+            device=device,
+        )
+
+    with torch.device("meta"):
+        model = model_fn()
+
+    # Based on pp rank split the model into logical stages and get the logical stages
+    # for this rank
+    logical_stage_mods: dict[int, torch.nn.Module] = {
+        0: model,
+    }
+    # AutoParallel API call
+    stage_mods: dict[int, torch.nn.Module] = {}
+    stage_graphs: dict[int, GraphCallables] = {}
+    stage_graph_metas: dict[int, GraphMeta] = {}
+    for stage_idx, stage_mod in logical_stage_mods.items():
+        with AutoParallel(stage_mod, input_fn, mesh, dynamic=True) as autop:
```

Comment:
FWIW, we don't need to literally use AutoParallel here, SimpleFSDP would also be good too.